### DISCUSSION TITLE
SHA1 Key not being updated/read

### DISCUSSION THREAD

**Discussion by gg17777** [codemagic-ci-cd/codemagic-docs#2738] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Upload SHA 1 key 
2. Start a new Build
3. The Android App Bundle was signed with the wrong key.

### Expected results

Being able to upload the app to the stores

### Actual results

The Android App Bundle was signed with the wrong key.

The SHA1 key that is "expected" is uploaded but it keeps not being able to read it and creates a new key each time

**Comment by dtrdic** [comment]
Hi @gg17777 so, when you create a binary and upload it manually to Google play console this binary is signed with a specific SHA signature (which comes from a Keystore)
You can see this SHA signature under **Setup > App signing** section in google play console.
![image](https://github.com/user-attachments/assets/7fe137af-6936-4b98-a932-3f7f7d71e583)

Based on the error message you provided, it seems the uploaded binary was signed with a different Keystore than the one used in Codemagic. You can verify the SHA signature of your Keystore by running the following command in your terminal:
`​keytool -list -v -keystore your_keystore_file.keystore`

What i suggest is;
1. create a fresh new google play app (with different bundle id),
2. In Codemagic run a build to generate a binary 
3. **Manually** publish this new binary to GooglePlay
4. Re-run the build on Codemagic. Now the binary will be published with the correct signature 

OR
If you don't want to do that, you can request a signing key change from Google. Google will remove the current signature from the Play Store, allowing you to use the Keystore you’ve set up in Codemagic. You can submit the request [here](https://support.google.com/googleplay/android-developer/contact/key).

After submitting the request, your App Signing page will look like this: 
 
![image](https://github.com/user-attachments/assets/661959db-6d41-47a9-b4ed-09035c102e80)

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
